core/src/test/kotlin/model/paginator/PaginatorTest.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2022 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package model.paginator
+
+import com.expediagroup.sdk.core.model.Response
+import com.expediagroup.sdk.core.model.paging.Paginator
+import com.expediagroup.sdk.core.model.paging.ResponsePaginator
+import com.expediagroup.sdk.core.test.ClientFactory.createExpediaGroupClient
+import com.expediagroup.sdk.core.test.TestConstants.SUCCESSFUL_DUMMY_REQUEST
+import io.ktor.client.call.body
+import io.ktor.client.statement.HttpResponse
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+class PaginatorTest {
+    @Nested
+    inner class PaginatorTest {
+        @Test
+        fun `test paginator with one response`() {
+            val client = createExpediaGroupClient()
+            val firstResponse = Response(200, "first", emptyMap())
+            val getBody: suspend (HttpResponse) -> String = { it.body<String>() }
```

Comment:
It might be a good idea to define this variable in one place since it's shared between all tests